File: src/types/stats.ts
export interface ReadingStats {
  // View counts (view_abstract action)
  todayCount: number;         // Today's view count
  weekCount: number;          // Last 7 days view count
  days30Count: number;        // Last 30 days view count
  monthCount: number;         // This month's view count (calendar month)
  // Read counts (download action)
  todayReadCount: number;     // Today's read count
  weekReadCount: number;      // Last 7 days read count
  days30ReadCount: number;    // Last 30 days read count
  monthReadCount: number;     // This month's read count (calendar month)
  // Favorites
  weekFavorites: number;      // Last 7 days favorites
  days30Favorites: number;    // Last 30 days favorites
  totalFavorites: number;     // Total favorites
  // Chats
  weekChats: number;          // Last 7 days chats
  days30Chats: number;        // Last 30 days chats
  totalChats: number;         // Total chats
  // Average
  avgDailyCount: number;
}

export interface HourlyDistribution {
  hour: number;
  count: number;
}

export interface WeeklyHourData {
  day: string;
  dayIndex: number;
  hour: number;
  count: number;
}

export interface DailyHourData {
  date: string;
  hour: number;
  count: number;
}

export interface DomainDistribution {
  domain: string;
  count: number;
  percentage: number;
}

export interface KeywordItem {
  text: string;
  value: number;
}

export interface HeatmapCell {
  date: string;
  count: number;
  level: number; // 0-4 用于颜色等级
}

export interface StatsData {
  readingStats: ReadingStats;
  hourlyDistribution: HourlyDistribution[];
  weeklyHourData: WeeklyHourData[];
  dailyHourData: DailyHourData[];
  domainDistribution: DomainDistribution[];
  keywords: KeywordItem[];
  heatmapData: HeatmapCell[];
}

File: src-tauri/src/models/stats.rs
// Stats models

use serde::{Deserialize, Serialize};

/// Reading statistics
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReadingStats {
    // View counts (view_abstract action)
    pub today_count: i64,        // Today's view count
    pub week_count: i64,         // Last 7 days view count
    pub days_30_count: i64,      // Last 30 days view count
    pub month_count: i64,        // This month's view count (calendar month)
    // Read counts (download action)
    pub today_read_count: i64,   // Today's read count
    pub week_read_count: i64,    // Last 7 days read count
    pub days_30_read_count: i64, // Last 30 days read count
    pub month_read_count: i64,   // This month's read count (calendar month)
    // Favorites
    pub week_favorites: i64,     // Last 7 days favorites
    pub days_30_favorites: i64,  // Last 30 days favorites
    pub total_favorites: i64,    // Total favorites
    // Chats
    pub week_chats: i64,         // Last 7 days chats
    pub days_30_chats: i64,      // Last 30 days chats
    pub total_chats: i64,        // Total chats
    // Average
    pub avg_daily_count: f64,
}

/// Hourly distribution
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HourlyDistribution {
    pub hour: i32,
    pub count: i64,
}

/// Weekly hour data
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WeeklyHourData {
    pub day: String,
    pub day_index: i32,
    pub hour: i32,
    pub count: i64,
}

/// Daily hour data
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DailyHourData {
    pub date: String,
    pub hour: i32,
    pub count: i64,
}

/// Domain distribution
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DomainDistribution {
    pub domain: String,
    pub count: i64,
    pub percentage: f64,
}

/// Keyword data for word cloud
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KeywordData {
    pub text: String,
    pub value: i64,
}

/// Heatmap data
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HeatmapData {
    pub date: String,
    pub count: i64,
    pub level: i32,
}

/// Full stats response
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StatsResponse {
    pub reading_stats: ReadingStats,
    pub hourly_distribution: Vec<HourlyDistribution>,
    pub weekly_hour_data: Vec<WeeklyHourData>,
    pub daily_hour_data: Vec<DailyHourData>,
    pub domain_distribution: Vec<DomainDistribution>,
    pub keywords: Vec<KeywordData>,
    pub heatmap_data: Vec<HeatmapData>,
}